COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -1,82 +1,671 @@
+        Accum_capacitance_list.append(Accum_capacitance_table)
+        Accum_capacitance_normalized_list.append(Accum_capacitance_normalized_table)
+        Tox_list.append(Tox_table)
+        Nox_list.append(Nox_table)
+
+        # Add text
+        text = hv.Text(10, 0.00000000065, 'Flat band voltage_fit_2nd derivative: {} V \n'
+                                            'Flat band voltage first derivative: {} V \n'
+                                            'C accumulation: {} F \n'
+                                            'C accumulation/A: {} F/cm\N{SUPERSCRIPT TWO} \n'
+                                            'Tox: {} nm \n'
+                                            'Nox: {} cm\N{SUPERSCRIPT MINUS}\N{SUPERSCRIPT TWO}'.format(
+            np.round(np.median(flatband_voltage[0]), 2),
+            np.round(voltage_value_of_max_firstder, 2),
+            np.round(Accum_capacitance, 10),
+            np.round(Accum_capacitance_normalized, 10),
+            np.round(Tox, 2),
+            np.format_float_scientific(Nox, 2))).opts(style=dict(text_font_size='25pt'))
+
+        # If more than one file do not do the derivates plots
+        if not len(cv_files) == 1:
+            returnPlot = plot
+            returnPlot = customize_plot(returnPlot, "1C2", configs["IV_PQC"], **addConfigs)
+            return returnPlot, flatband_voltage[0], Accum_capacitance_table, Accum_capacitance_normalized_table, Tox_table, Nox_table
+
+        elif len(cv_files) == 1:
+            # Plot a vertical line in the value of the fb voltage
+            vline = hv.VLine(flatband_voltage[0]).opts(color='black', line_width=1.0)
+
+            # Plots of the derivatives
+            secondDerivativePlot = self.basePlots5.Curve.secondderivative
+
+            # Plots of the fits
+            right_line = hv.Curve(right_line).opts(color='blue',line_width=1.0)
+            fit_line = hv.Curve(fit_line).opts(color='red', line_width=1.5)
+            returnPlot = plot * right_line * fit_line * secondDerivativePlot * vline
+            returnPlot = customize_plot(returnPlot, "1C2", configs["IV_PQC"], **addConfigs)
+            returnplot2 = plot * fit_line * right_line * vline * text
+            return returnPlot, flatband_voltage[0], Accum_capacitance_table, Accum_capacitance_normalized_table, Tox_table, Nox_table, returnplot2
+
+    def find_full_depletion_c2(self, plot, data, configs, diode_files, **addConfigs):
+        """
+        Finds the full depletion voltage of all data samples and adds a vertical line for the full depletion in the
+        plot. Vertical line is the mean of all measurements. Furthermore, adds a text with the statistics.
+        :param plot: The plot object
+        :param data: The data files
+        :param configs: the configs
+        :param **addConfigs: the configs special for the 1/C2 plot, it is recomended to pass the same options here again, like in the original plot!
+        :return: The updated plot
+        """
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Global variable 'df' undefined at the module level](https://app.codacy.com/manual/dominic.bloech/UniDAQ/pullRequest?prid=5481807)